docs/content/getting_started/shopping_list_app.md

```diff
@@ -1,3 +1,1199 @@
+
+    ```typescript
+    import { DeleteItemCommand, DynamoDBClient } from '@aws-sdk/client-dynamodb';
+    import {
+        deleteShoppingListHandler,
+        DeleteShoppingListChainedHandlerFunction,
+        INTERCEPTORS,
+        Response,
+        LoggingInterceptor,
+    } from 'myapi-typescript-runtime';
+
+    const dbclient = new DynamoDBClient({ region: process.env.AWS_REGION });
+
+    /**
+     * Type-safe handler for the DeleteShoppingList operation
+     */
+    export const deleteShoppingList: DeleteShoppingListChainedHandlerFunction = async (request) => {
+        LoggingInterceptor.getLogger(request).info(
+            'Start DeleteShoppingList Operation',
+        );
+
+        const shoppingListId = request.input.requestParameters.shoppingListId;
+        await dbclient.send(
+        new DeleteItemCommand({
+            TableName: 'shopping_list',
+            Key: {
+                shoppingListId: {
+                    S: shoppingListId,
+                },
+            },
+        }),
+        );
+
+        return Response.success({
+            shoppingListId,
+        });
+    };
+
+    /**
+     * Entry point for the AWS Lambda handler for the DeleteShoppingList operation.
+     * The deleteShoppingListHandler method wraps the type-safe handler and manages marshalling inputs and outputs
+     */
+    export const handler = deleteShoppingListHandler(
+        ...INTERCEPTORS,
+        deleteShoppingList,
+    );
+    ```
+
+=== "get-shopping-lists.ts"
+    
+    *This logic either calls the Scan or Query command depending on the presense of a shoppingListId request parameter. It also handles pagination based on the presense of a pageSize and/or nextToken. The shoppingItems are stored as a serialized JSON string in the table/*
+
+    ```typescript
+    import { DynamoDBClient, QueryCommand, ScanCommand } from '@aws-sdk/client-dynamodb';
+    import {
+        getShoppingListsHandler,
+        GetShoppingListsChainedHandlerFunction,
+        INTERCEPTORS,
+        Response,
+        LoggingInterceptor,
+        ShoppingList,
+    } from 'myapi-typescript-runtime';
+
+    const dbclient = new DynamoDBClient({ region: process.env.AWS_REGION });
+
+    /**
+     * Type-safe handler for the GetShoppingLists operation
+     */
+    export const getShoppingLists: GetShoppingListsChainedHandlerFunction = async (request) => {
+        LoggingInterceptor.getLogger(request).info('Start GetShoppingLists Operation');
+
+        const nextToken = request.input.requestParameters.nextToken;
+        const pageSize = request.input.requestParameters.pageSize;
+        const shoppingListId = request.input.requestParameters.shoppingListId;
+        const Command = shoppingListId ? QueryCommand : ScanCommand;
+
+        const response = await dbclient.send(new Command({
+            TableName: 'shopping_list',
+            ConsistentRead: true,
+            Limit: pageSize ? Number(pageSize) : undefined,
```

Comment:
Nit: this should now come through as a number already so no need for the conversion :)

Reply:
Updated - Thanks